src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -137,13 +137,7 @@ public static function getDefaultIndex(ContainerBuilder $container, string $serv
             throw new InvalidArgumentException(implode('be public', $message));
         }
 
-        $defaultIndex = $rm->invoke(null);
-
-        if (!\is_string($defaultIndex)) {
-            throw new InvalidArgumentException(implode(sprintf('return a string (got "%s")', get_debug_type($defaultIndex)), $message));
-        }
-
-        return $defaultIndex;
+        return (string) $rm->invoke(null);
```

Comment:
I would not unconditionally cast to string here. that could have weird effects with other things than integers that you want to support here.

Reply:
```php
if (\is_int($defaultIndex)){
    $defaultIndex = (string) $defaultIndex;
}
if (!\is_string($defaultIndex)) {
....
```
Is that better?